scripts/website/expand_templates/linters.go

```diff
@@ -0,0 +1,315 @@
+
+	for i := 0; i < tp.NumField(); i++ {
+		if strings.EqualFold(name, tp.Field(i).Name) {
+			return true
+		}
+	}
+
+	return false
+}
+
+func span(title, icon string) string {
+	return fmt.Sprintf(`<span title=%q>%s</span>`, title, icon)
+}
+
+func spanWithID(id, title, icon string) string {
+	return fmt.Sprintf(`<span id=%q title=%q>%s</span>`, id, title, icon)
+}
+
+type SettingSnippets struct {
+	ConfigurationFile string
+	LintersSettings   string
+}
+
+func extractExampleSnippets(example []byte) (*SettingSnippets, error) {
+	var data yaml.Node
+	err := yaml.Unmarshal(example, &data)
+	if err != nil {
+		return nil, err
+	}
+
+	root := data.Content[0]
+
+	globalNode := &yaml.Node{
+		Kind:        root.Kind,
+		Style:       root.Style,
+		Tag:         root.Tag,
+		Value:       root.Value,
+		Anchor:      root.Anchor,
+		Alias:       root.Alias,
+		HeadComment: root.HeadComment,
+		LineComment: root.LineComment,
+		FootComment: root.FootComment,
+		Line:        root.Line,
+		Column:      root.Column,
+	}
+
+	snippets := SettingSnippets{}
+
+	builder := strings.Builder{}
+
+	for j, node := range root.Content {
+		switch node.Value {
+		case "run", "output", "linters", "linters-settings", "issues", "severity":
+		default:
+			continue
+		}
+
+		nextNode := root.Content[j+1]
+
+		newNode := &yaml.Node{
+			Kind: nextNode.Kind,
+			Content: []*yaml.Node{
+				{
+					HeadComment: fmt.Sprintf("See the dedicated %q documentation section.", node.Value),
+					Kind:        node.Kind,
+					Style:       node.Style,
+					Tag:         node.Tag,
+					Value:       "option",
+				},
+				{
+					Kind:  node.Kind,
+					Style: node.Style,
+					Tag:   node.Tag,
+					Value: "value",
+				},
+			},
+		}
+
+		globalNode.Content = append(globalNode.Content, node, newNode)
+
+		if node.Value == "linters-settings" {
+			snippets.LintersSettings, err = getLintersSettingSections(node, nextNode)
+			if err != nil {
+				return nil, err
+			}
+
+			_, _ = builder.WriteString(
+				fmt.Sprintf(
+					"### `%s` configuration\n\nSee the dedicated [linters-settings](/usage/linters) documentation section.\n\n",
+					node.Value,
+				),
+			)
+			continue
+		}
+
+		nodeSection := &yaml.Node{
+			Kind:    root.Kind,
+			Style:   root.Style,
+			Tag:     root.Tag,
+			Value:   root.Value,
+			Content: []*yaml.Node{node, nextNode},
+		}
+
+		snippet, errSnip := marshallSnippet(nodeSection)
+		if errSnip != nil {
+			return nil, errSnip
+		}
+
+		_, _ = builder.WriteString(fmt.Sprintf("### `%s` configuration\n\n%s", node.Value, snippet))
+	}
+
+	overview, err := marshallSnippet(globalNode)
+	if err != nil {
+		return nil, err
+	}
+
+	snippets.ConfigurationFile = overview + builder.String()
+
+	return &snippets, nil
+}
+
+func getLintersSettingSections(node, nextNode *yaml.Node) (string, error) {
+	linters, err := readJSONFile[[]*shared.LinterWrapper](filepath.Join("assets", "linters-info.json"))
+	if err != nil {
+		panic(err)
```

Comment:
Can be:
```suggestion
		return "", err
```

Reply:
In this context, an error or panic is the same thing.
This method is used to fill a map form the template, it's easier to not have to handle error.